kodi: implement `kodiWithPlugins`, add more plugins

oxij opened the pull request:
###### Motivation for this change

I tried to add YATP to Kodi, ended up rewriting a bunch of stuff.

###### Known issues

I did not touch `wrapKodi` and the default wrapped `kodi`, but, really, we should probably deprecate and then remove it.

###### Things done

- [X] It builds.
- [X] Both old `kodi` with its configuration via `nixpkgs.config` and the new `kodiWithPlugins` work.

<strike>Don't merge sooner than 2018-09-09 23:59 UTC.</strike> Merge at will.

oxij committed:
kodiPlugins.steam-launcher: fix what is clearly a bug

oxij committed:
kodi: wrapper: simplify

oxij committed:
kodi: passthu pythonPackages

oxij committed:
kodi: wrapper: set PYTHONPATH from plugins

oxij committed:
kodiPlugins: inline commons.nix

oxij committed:
kodiPlugins: simplify, generalize, implement kodiWithPlugins

oxij commented:
/cc @titanous @edwtjo @cpages @worldofpeace from git-blame of `kodiPlugins`
/cc @edolstra because I reverted a tiny piece of yours here

GrahamcOfBorg commented:
<!--REQUEST_ID=a0566134-f419-4c1f-9d63-4ec88e4a1ea1-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46196&attempt_id=cb9d6359-61cd-4e0e-a926-22a0b2a29fa4)

Attempted: kodi, kodiPlugins.simpleplugin, kodiPlugins.yatp

The following builds were skipped because they don't evaluate on x86_64-linux: kodiPlugins.steam-launcher

<details><summary>Partial log (click to expand)</summary><p>

```
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/0v7q2iifi2j10yfln9bkfa9h9j9ibcsy-kodi-plugin-yatp-3.3.2
patching script interpreter paths in /nix/store/0v7q2iifi2j10yfln9bkfa9h9j9ibcsy-kodi-plugin-yatp-3.3.2
checking for references to /build in /nix/store/0v7q2iifi2j10yfln9bkfa9h9j9ibcsy-kodi-plugin-yatp-3.3.2...
/nix/store/n90qpxp56f81k6s94hakrbndwv8c0flc-kodi-with-plugins-17.6
/nix/store/h51bg87wk936qlm7d0n53k165aif6zbg-kodi-plugin-simpleplugin-2.3.2
/nix/store/0v7q2iifi2j10yfln9bkfa9h9j9ibcsy-kodi-plugin-yatp-3.3.2
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=a0566134-f419-4c1f-9d63-4ec88e4a1ea1-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46196&attempt_id=9166b1d5-caf1-462d-b8ad-cf3d796160e8)

Attempted: kodiPlugins.simpleplugin, kodiPlugins.yatp

The following builds were skipped because they don't evaluate on aarch64-linux: kodi, kodiPlugins.steam-launcher

<details><summary>Partial log (click to expand)</summary><p>

```
no configure script, doing nothing
building
no Makefile, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/fy0xvhfd5zk7gmrllwzcxhgqjqrh9jhm-kodi-plugin-yatp-3.3.2
patching script interpreter paths in /nix/store/fy0xvhfd5zk7gmrllwzcxhgqjqrh9jhm-kodi-plugin-yatp-3.3.2
checking for references to /build in /nix/store/fy0xvhfd5zk7gmrllwzcxhgqjqrh9jhm-kodi-plugin-yatp-3.3.2...
/nix/store/jvxdq68dlcy878gd0gppkd84nfalf65w-kodi-plugin-simpleplugin-2.3.2
/nix/store/fy0xvhfd5zk7gmrllwzcxhgqjqrh9jhm-kodi-plugin-yatp-3.3.2
```
</p></details>

oxij commented:
@xeji

Ugh, I see, `pythonPackages.requests` is a mass rebuilt.

I don't want to move this to staging as I rebased this several times already, it touches a lot of kodi stuff and I'm tired of fixing those conflicts.

I removed YATP from here, changes to plugins will go to `staging` after this gets merged to `master`.

@worldofpeace

> Please drop this plugin as it no longer exists.

Ok, will do in that second PR.

This is now became a purely infrastructure PR.

GrahamcOfBorg commented:
<!--REQUEST_ID=8db02893-ecbb-4d25-9adf-34beee88e0e7-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46196&attempt_id=8a4ebe50-43c4-4328-a453-15eb50e6a938)

The following builds were skipped because they don't evaluate on aarch64-linux: kodi, kodiPlugins.steam-launcher

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8db02893-ecbb-4d25-9adf-34beee88e0e7-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46196&attempt_id=030e411f-b39b-4bea-b70f-e69696a755fe)

Attempted: kodi

The following builds were skipped because they don't evaluate on x86_64-linux: kodiPlugins.steam-launcher

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/l7l9j7y4haayli2qw0fs90w6j8kg4m90-bluez-5.50' from 'https://cache.nixos.org'...
copying path '/nix/store/5qykkwx1xx2zsr3vnirhvw60ahrn91iw-xf86miscproto-0.9.3' from 'https://cache.nixos.org'...
copying path '/nix/store/76xzpl5q63y2cf2xrl8q8hfh03dgsy55-cwiid-2010-02-21-git' from 'https://cache.nixos.org'...
copying path '/nix/store/x40l832fzx6xjlwczhlddcnfjaqvv0pm-libXxf86misc-1.0.3' from 'https://cache.nixos.org'...
copying path '/nix/store/hmx9ryz40xxhf60g18mx199qcvi7b2gr-yajl-2.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/js6ll6kcxwcjid92zhwwsd508cvvv7rn-xdpyinfo-1.3.2' from 'https://cache.nixos.org'...
copying path '/nix/store/v7kr9w76qfhm5h7hchvarw1k95v4b7ws-kodi-17.6' from 'https://cache.nixos.org'...
building '/nix/store/j27ma3apd2zyjdpv9najrasqmsqd6hgx-kodi-with-plugins-17.6.drv'...
created 6 symlinks in user environment
/nix/store/n90qpxp56f81k6s94hakrbndwv8c0flc-kodi-with-plugins-17.6
```
</p></details>

oxij commented:
I've rebuilt everything (with mass-rebuild YATP patches included) on top of this revision. I can see no issues.

LGTM. Merge at will.

oxij commented:
@xeji Thanks!

oxij commented:
Changes to plugins: #46403.
